jobs/openshift/cluster/integreatly/test/Jenkinsfile

```diff
@@ -0,0 +1,109 @@
+      [$class: 'BooleanParameterValue', name: 'dryRun', value: params.dryRun]
+    ]
+    buildJob(jobName, jobParams)
+  }
+
+  stage('Install Integreatly') {
+    println "[INFO] Installing Integreatly in ${testOptions.clusterName} cluster"
+    def jobName = 'openshift-cluster-integreatly-install'
+    def userCount = '50'
+    def selfSignedCerts = false
+    def jobParams = [
+      [$class: 'StringParameterValue', name: 'clusterName', value: testOptions.clusterName],
+      [$class: 'StringParameterValue', name: 'openshiftMasterUrl', value: testOptions.openshiftMasterUrl],
+      [$class: 'StringParameterValue', name: 'clusterAdminUsername', value: testOptions.adminUsername],
+      [$class: 'StringParameterValue', name: 'clusterAdminPassword', value: testOptions.adminPassword],
+      [$class: 'StringParameterValue', name: 'installerGitUrl', value: testOptions.gitUrl],
+      [$class: 'StringParameterValue', name: 'installerGitBranch', value: testOptions.gitBranch],
+      [$class: 'StringParameterValue', name: 'userCount', value: userCount],
+      [$class: 'BooleanParameterValue', name: 'selfSignedCerts', value: selfSignedCerts],
+      [$class: 'BooleanParameterValue', name: 'dryRun', value: params.dryRun]
+    ]
+    buildJob(jobName, jobParams)
+  }
+
+  stage('Test Integreatly') {
+    println "[INFO] Testing Integreatly"
+  }
+
+  stage('Uninstall Integreatly') {
+    println "[INFO] Uninstalling Integreatly from ${testOptions.clusterName}"
+    def jobName = 'openshift-cluster-integreatly-uninstall'
+    def jobParams = [
+      [$class: 'StringParameterValue', name: 'clusterName', value: testOptions.clusterName],
+      [$class: 'StringParameterValue', name: 'openshiftMasterUrl', value: testOptions.openshiftMasterUrl],
+      [$class: 'StringParameterValue', name: 'clusterAdminUsername', value: testOptions.adminUsername],
+      [$class: 'StringParameterValue', name: 'clusterAdminPassword', value: testOptions.adminPassword],
+      [$class: 'StringParameterValue', name: 'uninstallerGitUrl', value: testOptions.gitUrl],
+      [$class: 'StringParameterValue', name: 'uninstallerGitBranch', value: testOptions.gitBranch],
+      [$class: 'BooleanParameterValue', name: 'dryRun', value: params.dryRun]
+    ]
+    buildJob(jobName, jobParams)
+  }
+
+  stage('Deprovision Cluster') {
+    println "[INFO] Deprovision ${testOptions.clusterName} cluster"
+    def jobName = 'openshift-cluster-deprovision'
+    def jobParams = [
+      [$class: 'StringParameterValue', name: 'clusterName', value: testOptions.clusterName],
+      [$class: 'StringParameterValue', name: 'awsRegion', value: testOptions.awsRegion],
+      [$class: 'StringParameterValue', name: 'awsAccountName', value: testOptions.awsAccountName],
+      [$class: 'StringParameterValue', name: 'clusterDomainName', value: testOptions.domainName],
+      [$class: 'BooleanParameterValue', name: 'dryRun', value: params.dryRun]
+    ]
+    buildJob(jobName, jobParams)
+  }
```

Comment:
@JameelB We need to ensure this stage always gets executed at the end regardless of the status of any of the earlier stages. I think unfortunately the only way to do that in a scripted pipeline is wrapping everything a try finally block. If you felt like you could also try a declarative pipeline instead which provides you with a post/always option.